jerry-core/api/jerry-snapshot.c

```diff
@@ -520,6 +550,47 @@ jerry_snapshot_set_offsets (uint32_t *buffer_p, /**< buffer */
         }
       }
 
+#if ENABLED (JERRY_ES2015)
+      /* function name */
+      if (!(bytecode_p->status_flags & CBC_CODE_FLAGS_CLASS_CONSTRUCTOR))
+      {
+        if (literal_start_p[-1] != ECMA_VALUE_ANONYMOUS)
+        {
+          JERRY_ASSERT (ecma_is_value_string (literal_start_p[-1]));
+
+          lit_mem_to_snapshot_id_map_entry_t *current_p = lit_map_p;
+
+          while (current_p->literal_id != literal_start_p[-1])
+          {
+            current_p++;
+          }
+
+          literal_start_p[-1] = current_p->literal_offset;
+        }
+
+        literal_start_p--;
+      }
+#endif /* ENABLED (JERRY_ES2015) */
+
+#if ENABLED (JERRY_RESOURCE_NAME)
+      int32_t resource_name_index = (bytecode_p->status_flags & CBC_CODE_FLAG_HAS_TAGGED_LITERALS) ? -2 : -1;
```

Comment:
I was wondering if we could move this `? :` part to a macro as it is used at a few places, but I'm not sure if it is a good idea.

Reply:
I'd keep them like this for now, I'm planning to improve the snapshot tool later so maybe this could be improved as well.